File: backend/app/schemas/conflicts.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel


ConflictType = Literal[
    "FORWARD_MISMATCH",
    "PTR_MISMATCH",
    "NO_FORWARD",
    "DUPLICATE_HOSTNAME",
]


class ConflictItem(BaseModel):
    ip_address: str
    hostname: str
    conflict_type: ConflictType
    detail: str


class ConflictReport(BaseModel):
    subnet_id: str
    scanned_at: datetime
    total_checked: int
    conflicts: list[ConflictItem]

File: frontend/src/types/conflicts.ts
export type ConflictType =
  | 'FORWARD_MISMATCH'
  | 'PTR_MISMATCH'
  | 'NO_FORWARD'
  | 'DUPLICATE_HOSTNAME';

export interface ConflictItem {
  ip_address: string;
  hostname: string;
  conflict_type: ConflictType;
  detail: string;
}

export interface ConflictReport {
  subnet_id: string;
  scanned_at: string;
  total_checked: number;
  conflicts: ConflictItem[];
}
